flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/partition/PartitionTableTest.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.taskexecutor.partition;
+
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.io.network.partition.ResultPartitionID;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import static junit.framework.TestCase.assertNotNull;
+import static junit.framework.TestCase.assertTrue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Tests for the {@link PartitionTable}.
+ */
+public class PartitionTableTest extends TestLogger {
+
+	private static final JobID JOB_ID = new JobID();
+	private static final ResultPartitionID PARTITION_ID = new ResultPartitionID();
+
+	@Test
+	public void testEmptyTable() {
+		final PartitionTable table = new PartitionTable();
+
+		// an empty table should always return an empty collection
+		Collection<ResultPartitionID> partitionsForNonExistingJob = table.stopTrackingPartitions(new JobID());
+		assertNotNull(partitionsForNonExistingJob);
+		assertEquals(0, partitionsForNonExistingJob.size());
+
+		assertFalse(table.hasTrackedPartitions(new JobID()));
```

Comment:
this job id should be the same with above `table.stopTrackingPartitions`, so might use `JOB_ID` instead.

Reply:
IT doesn't have to be the same, but it might as well be using JOB_ID.

Reply:
Yes, any job id is not be tracked in table before, so no need to be the same.